### DISCUSSION TITLE
Is there a podman API which can be used for doing podman login to a private registry?

### DISCUSSION THREAD

**Discussion by sriram241294** [containers/podman#18868] [Q&A] [answered]
0

I have an existing application deployed in podman. I want to do a podman login into a private registry from inside the container so that i will be able to pull/push images into the private registry. I'm trying to find a podman api for it but didn't fumble upon one till now.

I'm trying it in podman version 4.2.0 in rhel 9.1.

I'm not able to find the api in the [link](https://docs.podman.io/en/latest/_static/api.html#tag/images/operation/ImagePullLibpod) also.

**Comment by vrothberg** [comment, marked as answer]
Thank you for reaching out, @sriram241294.

In the link, you can see the `X-Registry-Auth` header mentioned.  This is where you can specify credentials.  There is nt endpoint to "login" on the server side since clients need to provide the credentials via the http-request header.
